contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,743 @@
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     * @param yieldTokenBalance is the collateral token (stETH) balance of the vault
+     */
+    function withdrawBaseToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount,
+        uint256 yieldTokenBalance
+    ) internal {
+        uint256 underlyingTokensToWithdraw =
+            IWSTETH(collateralToken).getStETHByWstETH(
+                withdrawAmount.sub(yieldTokenBalance)
+            );
+
+        transferAsset(payable(recipient), underlyingTokensToWithdraw);
+    }
+
+    /**
+     * @notice Unwraps the necessary amount of the yield-bearing yearn token
+     *         and transfers amount to vault
+     * @param amount is the amount of `asset` to withdraw
+     * @param collateralToken is the address of the collateral token
+     * @param crvPool is the address of the steth <-> eth pool on curve
+     * @param minETHOut is the min eth to recieve
+     */
+    function unwrapYieldToken(
+        uint256 amount,
+        address collateralToken,
+        address crvPool,
+        uint256 minETHOut
+    ) external {
+        uint256 assetBalance = address(this).balance;
+
+        uint256 amountToUnwrap =
+            IWSTETH(collateralToken).getWstETHByStETH(
+                dsmax(assetBalance, amount).sub(assetBalance)
+            );
+
+        if (amountToUnwrap > 0) {
+            IWSTETH wsteth = IWSTETH(collateralToken);
+            // Unrap to stETH
+            wsteth.unwrap(amountToUnwrap);
+
+            ICRV crv = ICRV(crvPool);
+
+            // CRV SWAP HERE from steth -> eth
+            // 0 = ETH, 1 = STETH
+            crv.exchange(1, 0, amountToUnwrap, minETHOut);
+        }
+    }
+
+    /**
+     * @notice Wraps the necessary amount of the base token to the yield-bearing yearn token
+     * @param weth is the address of weth
+     * @param collateralToken is the address of the collateral token
+     */
+    function wrapToYieldToken(address weth, address collateralToken) external {
+        IWSTETH collateral = IWSTETH(collateralToken);
+        address steth = collateral.stETH();
+        IERC20 stethToken = IERC20(steth);
+
+        // Unwrap all weth premiums transferred to contract
+        IWETH weth = IWETH(weth);
+        uint256 wethBalance = weth.balanceOf(address(this));
+        if (wethBalance > 0) {
+            weth.withdraw(wethBalance);
+        }
+
+        if (address(this).balance > 0) {
+            // Send eth to Lido, recieve steth
+            ISTETH(steth).submit{value: address(this).balance}(address(this));
+            uint256 stethBalance = stethToken.balanceOf(address(this));
+            // approve wrap
+            stethToken.doubleApprove(collateralToken, stethBalance.add(1));
+            // Wrap to wstETH - need to add 1 to steth balance as it is innacurate
+            collateral.wrap(stethBalance.add(1));
```

Comment:
for some reason stethBalance is off by 1 wei. so adding one to wrap. (I checked steth balance after wrap and it was 1)